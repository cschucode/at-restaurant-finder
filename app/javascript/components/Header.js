import React from 'react';

const Header = () => {
  return (
    <div>
      <div>Logo with Text</div>
      <button>Filter</button>
      <input type="text" placeholder="Find a Restaurant" />
    </div>
  );
}

export default Header;