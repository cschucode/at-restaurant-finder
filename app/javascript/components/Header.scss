.header {
  align-items: center;
  border-bottom: none;
  box-shadow: 0 1px 2px #bbb;
  display: flex;
  flex-direction: row;
  height: 60px;
  padding: 0 30px;


  &__logo {
    align-items: center;
    display: flex;
    height: 100%;

    span {
      font-size: 24px;
    }
  }

  &__logo-image {
    height: 100%;
  }

  &__search {
    align-items: center;
    display: flex;
    flex: 1;
    justify-content: flex-end;
  }

  &__filter-btn,
  &__search-input {
    background-color: #fff;
    border: 1px solid #bbb;
    border-radius: 3px;
    font-size: 12px;
    font-weight: 100;
    margin-right: 15px;
    padding: 6px 12px;
  }

  &__search-input {
    border-right: none;
    margin-right: 0;
    outline: none;
  }

  &__search-btn {
    background-color: #fff;
    background-image: url('../../assets/images/search.png');
    background-repeat: no-repeat;
    background-size: cover;
    border: 1px solid #bbb;
    border-left: none;
    height: 28px;
    width: 28px;
  }
}