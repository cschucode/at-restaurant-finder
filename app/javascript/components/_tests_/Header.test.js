import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header';

describe('<Header />', () => {
  test('Render Header component', () => {
    const component = renderer.create(
      <Header handleSearch={jest.fn()} handleSortClick={jest.fn()} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
