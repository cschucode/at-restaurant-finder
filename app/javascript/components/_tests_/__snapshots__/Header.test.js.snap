// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Render Header component 1`] = `
<header
  className="header"
>
  <div
    className="header__logo"
  >
    <img
      alt="All Trails Logo"
      className="header__logo-image"
      src="https://cdn-assets.alltrails.com/assets/shared/AT_Main_Logo-da354e8947c2636e86fb27f60cbb037b.png"
    />
    <span>
      at Lunch
    </span>
  </div>
  <div
    className="header__search"
  >
    <div
      className="header__sort"
    >
      <button
        className="header__sort-btn"
        onClick={[Function]}
      >
        Sort
      </button>
      
    </div>
    <input
      className="header__search-input"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Search for a restaurant"
      type="text"
      value=""
    />
    <button
      className="header__search-btn"
      onClick={[Function]}
    />
  </div>
</header>
`;
